### PR TITLE
CLOUDP-71334: Fix MongoDB 4.4 Scaling Up

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

With MongoDB 4.4 it is no longer possible to scale up more than one member at a time (besides the first time creation of the deployment)

This PR brings the scale package from the enterprise code base and implements it for the community MongoDB resource

### CODE REVIEW

File: pkg/util/scale/scale.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+func IsStillScaling(replicaSetScaler ReplicaSetScaler) bool {
+	return ReplicasThisReconciliation(replicaSetScaler) != replicaSetScaler.DesiredReplicaSetMembers()
+}
+
+func isScalingDown(replicaSetScaler ReplicaSetScaler) bool {
+	return replicaSetScaler.DesiredReplicaSetMembers() < replicaSetScaler.CurrentReplicaSetMembers()
+}
+
+// AnyAreStillScaling reports true if any of one the provided members is still scaling
```
Comment (rodrigovalin) [inline pkg/util/scale/scale.go:35]: Is this function used?

File: pkg/apis/mongodb/v1/mongodb_types.go [hunk 2]
```diff
@@ -218,6 +220,18 @@ type MongoDB struct {
...
+func (m MongoDB) DesiredReplicaSetMembers() int {
+	return m.Spec.Members
+}
+
+func (m MongoDB) CurrentReplicaSetMembers() int {
+	return m.Status.Members
+}
+
+func (m *MongoDB) MembersThisReconciliation() int {
+	return scale.ReplicasThisReconciliation(m)
```
Comment (rodrigovalin) [inline pkg/apis/mongodb/v1/mongodb_types.go:232]: `Members` or `Replicas` ? What about agreeing on 1 term?

**Review by rodrigovalin — COMMENTED** [review]
LGTM, straightforward implementation! ❤️ 

Can we agree on the term used to refer to the "elements of the set of mongodb on a replicaset" as "replicas" or "members"? Or do you think both terms should be used in different contexts?

File: pkg/util/scale/scale.go (hunk 1 above)
Comment (chatton) [reply pkg/util/scale/scale.go:35]: good catch, this is actually used in the enterprise operator for the ShardedCluster implementation, this package will be imported there.

File: pkg/apis/mongodb/v1/mongodb_types.go (hunk 2 above)
Comment (rodrigovalin) [reply pkg/apis/mongodb/v1/mongodb_types.go:232]: I vote for Replicas, replicas everywhere

**Comment by chatton** [comment]
@rodrigovalin let me know if you're happy with the rename!

**Review by rodrigovalin — APPROVED** [review]
LGTM

**Pull request merged**